src/proxy.ts

```diff
@@ -49,4 +49,30 @@ function computeApi3ReaderProxyV1Address(
   );
 }
 
-export { computeApi3ReaderProxyV1Address };
+function computeCommunalApi3ReaderProxyV1Address(chainId: ethers.BigNumberish, dapiName: string) {
+  return computeApi3ReaderProxyV1Address(chainId, dapiName, 1, '0x');
+}
+
+function computeDappId(dappAlias: string, chainId: ethers.BigNumberish) {
+  if (!DAPPS.some((dapp) => dapp.alias === dappAlias)) {
+    throw new Error(`dApp with alias ${dappAlias} not registered to the package`);
+  }
+  // Chain ID is a string to also support non-numerical (i.e., non-EVM) chain IDs.
+  // We use the BigNumberish type for now, until we need to support more arbitrary strings.
+  return ethers.solidityPackedKeccak256(['string', 'string'], [dappAlias, chainId.toString()]);
+}
+
+function computeDappSpecificApi3ReaderProxyV1Address(
+  dappAlias: string,
+  chainId: ethers.BigNumberish,
+  dapiName: string
+) {
+  return computeApi3ReaderProxyV1Address(chainId, dapiName, computeDappId(dappAlias, chainId), '0x');
+}
+
+export {
+  computeApi3ReaderProxyV1Address,
```

Comment:
Leaving this to not break the market, etc., and also for more flexible usage. The end users are only supposed to use `computeCommunalApi3ReaderProxyV1Address()` and `computeDappSpecificApi3ReaderProxyV1Address()` though.